mypy/constraints.py

```diff
@@ -258,9 +276,12 @@ def any_constraints(options: List[Optional[List[Constraint]]], eager: bool) -> L
         # every option, combine the bounds with meet/join always, not just for Any.
         trivial_options = select_trivial(valid_options)
         if trivial_options and len(trivial_options) < len(valid_options):
-            # Randomly choose first trivial option for source of Any.
-            # TODO: is it possible to get more precise attribution of source Any?
-            any_by_type_var = {c.type_var: c.target for c in trivial_options[0]}
+            # TODO: if we will support multiple source Any, use this here.
+            # For now we chose some arbitrary Any in a stable way.
+            any_by_type_var = {
+                tv: sorted(anys, key=get_any_kind)[0]
+                for tv, anys in collect_target_anys(trivial_options).items()
+            }
```

Comment:
If this gets executed lot, this could be fairly expensive. We could easily add a fast path for common simple cases. However, if this is executed fairly infrequently, it doesn't make sense to optimize things. Do you have an idea of how frequently we'll get here?